Source/Modules/php.cxx

```diff
@@ -1193,10 +1203,17 @@ class PHP : public Language {
       Printf(f->code, "%s\n",magic_get);
       Printf(f->code, "\nelse if (strcmp(arg2->val,\"thisown\") == 0)\n{\n");
       Printf(f->code, "if(arg->newobject) {\nRETVAL_LONG(1);\n}\nelse {\nRETVAL_LONG(0);\n}\n}\n\n");
-      Printf(f->code, "else {\nif (!arg->extras) {\nRETVAL_NULL();\n}\n");
+      Printf(f->code, "else {\n");
+      if (baseClassExtend) {
+        Printf(f->code, "PHP_MN(%s___get)(INTERNAL_FUNCTION_PARAM_PASSTHRU);\n", baseClassExtend);
+        Printf(f->code, "if (Z_ISNULL_P(return_value) && arg->std.ce == SWIGTYPE_%s_ce) {\n", class_name);
```

Comment:
There are two reasons why `Z_ISNULL_P(return_value)` could be true here.  One is that the attribute isn't set on the parent class.  But the other reason is that it is set and its value is `NULL` - for example:

```
$ php7.0 -r '$a = new stdClass; $a->foo = NULL; var_dump($a);'
object(stdClass)#1 (1) {
  ["foo"]=>
  NULL
}
```

If an attribute is set to `42` via PHP it'll get set on the base class and also stored in the extras hash in the most derived class.  Then the C++ side updates the base class attribute to be `NULL`.  Then PHP read the attribute from the derived class, which fetches it from the base class, sees it is `NULL`, and returns `42` from the extras hash in the derived class instead, which is wrong - in this case `NULL` should be returned.

I continue to think that it would be better to just store "extra" attributes using a hash in the base class, and not ever store attributes twice.  If you store them twice, then you have to worry about something using the extra value when it's out of date.